modules/developers-guide/pages/design-apps.adoc

```diff
@@ -1,50 +1,49 @@
-However, if your application is retrieving sensitive information—like financial data—you might want more assurance about your results than a basic query provides.  
+That is a good general rule to follow and can be applied broadly to most categories of dapps.
+However, you should also consider the security and performance trade-off that queries don’t go through consensus and do not appear on the blockchain.
+For some dapps, that trade-off might be appropriate. For example, if you are developing a blogging platform, queries that retrieve articles matching a tag probably don’t warrant going through consensus to ensure that a majority of nodes agree on the results.
+However, if your dapp is retrieving sensitive information—like financial data—you might want more assurance about your results than a basic query provides.
 
-As an alternative to basic queries, the {IC} also supports *certified queries*. Certified queries enable you to receive *authenticated responses* that you can trust. Using certified queries is an advanced technique that is not covered in the tutorials or other developer-focused documentation, but you can learn about how the authentication works and what you need to do to configure your program to return certified data in response to queries in the link:../interface-spec/index{outfilesuffix}[Interface specification].
+As an alternative to basic queries, the {IC} also supports *certified queries*. Certified queries enable you to receive *authenticated responses* that end users can trust. Using certified queries is an advanced technique that is not covered in the tutorials or other developer-focused documentation, but you can learn about how the authentication works and what you need to do to configure your program to return certified data in response to queries in the link:../interface-spec/index{outfilesuffix}[Interface specification].
 
 == Data storage and retrieval
 
-For many applications, databases are primarily used to provide long term storage of records that can be retrieved efficiently using one or more key fields. On the {IC}, this use case is usually addressed through data structures like hash tables. 
-Instead of a traditional database, the {IC} enables you to use *stable memory* to handle long-term data storage—often referred to as orthogonal persistance—and to use *query calls* to retrieve your data.
-Although this approach works for most applications and simple key value database requests it doesn’t fully emulate a relational database.
+The {IC} enables you to use *stable memory* to handle long-term data storage—often referred to as orthogonal persistence—and to use *query calls* to retrieve your data.
+Efficiently retrieving data using one or more keys can typically be achieved by using data structures like hash tables.
+It is also possible to implement a more traditional database inside a canister.
```

Comment:
These feel like random examples. Maybe we should rather say that this of course depends on the application but ... and then give those as explicit examples?

Reply:
How about replacing the whole section with this?
```
The {IC} offers *orthogonal persistence*, which means that the full state of each canister is stored after every upgrade call.
You therefore do not have to explicitly persist data.
It is important to think about the data structures used, as this may affect how efficiently data can be retrieved.
For example, in the link:https://github.com/dfinity/linkedup[LinkedUp] sample dapp, a hash map is used to efficiently retrieve the profile of a given user id. 
```

Reply:
Sounds a lot better!

Reply:
thanks, committed that change